lms/djangoapps/grades/models.py

```diff
@@ -257,8 +257,14 @@ def save_grade(cls, **kwargs):
                     usage_key=usage_key,
                     defaults=kwargs,
                 )
+            log.info("persistent subsection grade successfully saved: {0}".format(unicode(grade)))
         except IntegrityError:
             cls.update_grade(user_id=user_id, usage_key=usage_key, **kwargs)
+            log.info("integrity error trying to save grade for user {0}, usage key {1}, defaults {2}".format(
```

Comment:
Actually, now that I'm looking at this closer, there's an issue with this call to `get_or_create`.  It should only pass in the field values that are used by the uniqueness constraints - otherwise, it will raise `IntegrityError`s whenever the subsection grade already exists and we're passing in different values for the `defaults`.  In fact, since `get_or_create` is atomic, we should not be getting an `IntegrityError` here and should remove this error handling - once we fix the call to `get_or_create`.